### DISCUSSION TITLE
Augmenting the library search path

### DISCUSSION THREAD

**Discussion by darrenjw** [google-research/dex-lang#972] [Q&A] [answered]
I've been enjoying playing around with Dex and have got to the point where I want to have programs spread across multiple source files. It's not obvious to me how to augment the library search path to make this simple. I know that there is a flag to change the lib directory, but if I change it, the prelude can't be found. I guess I could just copy the prelude and any other stuff I want from the standard library, but I wonder if I'm missing something? It would be good to keep the standard library and augment it with additional search paths.

**Comment by apaszke** [comment, marked as answer]
Agreed, this is definitely a missing feature. I guess we could just use the standard UNIX trick of using a list of library paths separated by colons. I'll see how easy it would be to support that.

**Reply by apaszke** [reply]
I just sent https://github.com/google-research/dex-lang/pull/975 that adds support for multiple library paths. The module system is still very young and undeveloped, so I expect there might be still quite a few rough edges. In particular, there's no support for hierarchical packages. So it might be ok for small- to medium-sized projects, but could get a bit unwieldy for larger things.

But, if you have use cases, we're happy to improve! There's a lot to be done when creating a new language, so we're fairly lazy when it comes to implementing new features. If you find any other things you'd like to see happen, let us know!

**Reply by darrenjw** [reply]
Excellent, thanks!

**Reply by apaszke** [reply]
As a side note, I only now realized that our readme instructions are a bit outdated. If you're not planning to be modifying the compiler, we actually now recommend building with `make build-opt`. It will make the compiler _much faster_. That is, unless you've used `make install`, which enables all Haskell optimizations by default as well.
